src/export-data/export-data.service.ts

```diff
@@ -0,0 +1,101 @@
+import { ForbiddenException, Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { Workbook, Worksheet } from 'exceljs';
+import { createTransport } from 'nodemailer';
+
+@Injectable()
+export class ExportDataService {
+  constructor(private prisma: PrismaService) {}
+
+  public async exportGameData(
+    userId: string,
+    emailTo: string,
+  ): Promise<string> {
+    const user = await this.prisma.user
+      .findUniqueOrThrow({
+        where: {
+          role: 'ADMIN',
+          id: userId,
+        },
+      })
+      .catch(() => {
+        throw new ForbiddenException(
+          'Only user with role admin can export the game data',
+        );
+      });
+
+    const histories = await this.prisma.gameHistory.findMany({
+      orderBy: {
+        gameType: 'asc',
+      },
+    });
+
+    try {
+      const { worksheet, workbook, transporter } = this.setup();
+      worksheet.columns = [
+        { header: 'Game Type', key: 'gameType' },
+        { header: 'Score', key: 'score' },
+        { header: 'Start Time', key: 'startTime' },
+        { header: 'End Time', key: 'endTime' },
+      ];
```

Comment:
untuk pembuatan worksheet.columns, mungkin bisa dipindah ke setup method, karena akan selalu sama columnsnya (?)

Reply:
okay to